tests/xdc/history_replication_dlq_test.go

```diff
@@ -96,23 +97,25 @@ type (
 	replicationTaskExecutorParams struct {
 		// We use an interface{} for these tasks because we don't care about the data, and the concrete type changes
 		// depending on whether we have streaming enabled for replication or not.
-		tasks chan interface{}
+		executedTasks       *channels.InfinitelyBufferedChannel[*replicationspb.ReplicationTask]
```

Comment:
It would work, but we'd have to know how much to buffer it by or just set a really high number and hope that's enough. It's a small thing, but a dynamic buffer size seemed better to me than trying to guess at a good buffer size. I moved it out of this PR though and just used 100